Make ARMC5 and IAR develop profile also size optimized

JanneKiiskila opened the pull request:
### Description

Due to some historical reasons ARMC 5 compiler behaves very
differently compared to others (GCC, IAR, ARM C 6) as it optimizes
performance rather than size. The compilers should behave the same
way with the same profile, thus ARM C 5 should also drive towards
size.

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

@kjbracey-arm @bulislaw  @0xc0170 

### Release Notes

Commit:
Make ARMC5 develop profile also size optimized

Due to some historical reasons ARMC 5 compiler behaves very
differently compared to others (GCC, IAR, ARM C 6) as it optimizes
performance rather than size (like the others).

All compilers should behave the same way with the same profile,
thus ARM C 5 should also drive towards size (space).

kjbracey reviewed (approved):
As far as I know, the original motivation for this was that ARM C 5 Otime produced similar code size to GCC 4 Ospace, and there was no point taking even less ROM than GCC, if we were testing all 3 toolchains in parallel - if GCC Ospace fitted then ARM C Otime would also fit.

GCC 6 and later produce much better ARM code, so ARM C needs Ospace to match it.

teetak01 reviewed (commented):
IAR could also use "-Ohz" instead.

I would also fix the Debug profile. With current Mbed OS profile, you cannot compile PDMC with Mesh stack using Mbed OS version of debug profile.

https://github.com/ARMmbed/mbed-cloud-client-example/blob/master/profiles/size.json
https://github.com/ARMmbed/mbed-cloud-client-example/blob/master/profiles/debug_size.json

bulislaw reviewed (approved):
Looks good to me, I'm assuming it doesn't cause troubles with binaries in the tree. Probably should go to 5.14, but I'm happy to be challenged here.

Commit:
Align also IAR - develop profile to be size optimized

IAR is also performance optimizing instead of size optimizing in
develop profile. Align also that (review feedback).
